test/models/role_test.rb

```diff
@@ -315,6 +315,19 @@ def test_allow_to_modify_roles_when_bypass_locking
         assert_empty @filter_with_org.organizations
       end
 
+      it 'should forced unlimited check if role or filter have no taxonomies' do
+        @role.organizations = [ @org1 ]
+        @role.save
+        @filter_with_org.reload
+        assert @role.filters.last.unlimited?
+        refute @role.filters.first.unlimited?
+        @role.organizations = []
+        @role.save
+        @filter_with_org.reload
+        assert @role.filters.last.unlimited?
```

Comment:
Not sure if the assertion for `.last` means anything?

Reply:
I think you could use `@filter_without_org` and `@filter_with_org` variables that are defined above. That deduplicates the logic.